ebs/ebs_service.go

```diff
@@ -542,6 +592,87 @@ func (s *ebsService) DetachVolume(volumeID string) error {
 	return detachErr
 }
 
+func (s *ebsService) GetMostRecentSnapshot(volumeName string, dcName string, filters ...*ec2.Filter) (*ec2.Snapshot, error) {
+	snapshots, err := s.GetSnapshots(volumeName, dcName, filters...)
+	if err != nil {
+		return nil, err
+	} else if len(snapshots) == 0 {
+		return nil, nil
+	}
+	return snapshots[len(snapshots)-1], nil
+}
+
+func (s *ebsService) GetMostRecentVolume(volumeName string, dcName string, filters ...*ec2.Filter) (*ec2.Volume, error) {
+		// We always need to specify the name, dc name, and that the snapshot is complete. If the AZ has truly crashed, then the snapshot may never complete
+	volumeInput := &ec2.DescribeVolumesInput {
+		Filters: []*ec2.Filter {
+			&ec2.Filter {
+				Name: aws.String("tag:Name"),
+				Values: []*string {
+					aws.String(volumeName),
+				},
+			},
+			&ec2.Filter {
+				Name: aws.String("tag:DCName"),
+				Values: []*string {
+					aws.String(dcName),
+				},
+			},
+			&ec2.Filter {
+				Name: aws.String("status"),
+				Values: []*string {
+					aws.String("available"),
+				},
+			},
+		},
+	}
+	volumeInput.Filters = append(volumeInput.Filters, filters...)
+	log.Printf("Describe volumes input: %+v", volumeInput)
+	req, volOutput := s.ec2Client.DescribeVolumesRequest(volumeInput)
+	if err := req.Send(); err != nil {
+		return nil, err
+	}
+	sort.Sort(VolumeByCreateTime(volOutput.Volumes))
+	if len(volOutput.Volumes) == 0 {
+		return nil, nil
+	}
+	return volOutput.Volumes[len(volOutput.Volumes)-1], nil
```

Comment:
this works but you can also use sort.Reverse to get a descending order of sorts. Ref: https://golang.org/pkg/sort/#Reverse